src/app/home/home.component.ts

```diff
@@ -14,23 +15,42 @@ import {HttpService} from '../http.service';
   styleUrls: ['./home.component.css'],
 })
 export class HomeComponent {
-  endpoint: string = '_recommend';
-  recommendation$: Observable<Recommendation>;
+  readonly recommendation$: Observable<Recommendation>;
+  readonly liked$: Observable<boolean>;
+
+  likeClicked = new Subject<Recommendation>();
 
   constructor(
+    public window: Window,
     public authService: AuthService,
     public httpService: HttpService
   ) {
     this.recommendation$ = authService.authenticateWithGoogle().pipe(
+      flatMap(() => authService.getIdToken()),
       flatMap((idToken: string) =>
-        httpService.get<Recommendation>(idToken, this.endpoint)
+        httpService.get<Recommendation>(idToken, '_recommend')
       ),
       map((song: Recommendation) => this.validateRecommendation(song))
     );
+    this.liked$ = this.likeClicked.pipe(this.likeRecommendation);
   }
 
   /** Validates/processes song recommendation. */
   validateRecommendation({album, ...rest}: Recommendation): Recommendation {
     return {album: album ? album : 'Non-Album Single', ...rest};
   }
+
+  /** Adds the recommended song to the user's liked songs. */
+  likeRecommendation: OperatorFunction<Recommendation, boolean> = pipe(
+    flatMap((song: Recommendation) =>
+      this.authService
+        .getIdToken()
+        .pipe(map((idToken: string) => [song, idToken]))
+    ),
+    flatMap(([song, idToken]: [Recommendation, string]) =>
+      this.httpService.post<Like>(idToken, song, '_like')
+    ),
+    map((like: Like) => !!like),
+    startWith(false)
```

Comment:
To make it clear what is starting with false, I would move this `startWith` to `this.liked$`

`this.likeClicked.pipe(this.likeRecommendation, startWith(false));`

Reply:
👍 that makes sense.